sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRowToColumnarExec.scala

```diff
@@ -480,27 +481,38 @@ private object GpuRowToColumnConverter {
     override def getNullSize: Double = childConverters.map(_.getNullSize).sum + VALIDITY
   }
 
-  //  private case class DecimalConverter(dt: DecimalType) extends TypeConverter {
-  //    override def append(row: SpecializedGetters,
-  //    column: Int,
-  //    builder: ai.rapids.cudf.HostColumnVector.Builder): Unit = {
-  //      if (row.isNullAt(column)) {
-  //        builder.appendNull()
-  //      } else {
-  //        val d = row.getDecimal(column, dt.precision, dt.scale)
-  //        if (dt.precision <= Decimal.MAX_INT_DIGITS) {
-  //          cv.appendInt(d.toUnscaledLong.toInt)
-  //        } else if (dt.precision <= Decimal.MAX_LONG_DIGITS) {
-  //          cv.appendLong(d.toUnscaledLong)
-  //        } else {
-  //          val integer = d.toJavaBigDecimal.unscaledValue
-  //          val bytes = integer.toByteArray
-  //          cv.appendByteArray(bytes, 0, bytes.length)
-  //        }
-  //      }
-  //    }
-  //  }
-  //
+  private class DecimalConverter(precision: Int, scale: Int) extends TypeConverter {
+
+    private val nonNullConverter = new NotNullDecimalConverter(precision, scale)
+
+    override def append(
+      row: SpecializedGetters,
+      column: Int,
+      builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
+      if (row.isNullAt(column)) {
+        builder.appendNull()
+      } else {
+        nonNullConverter.append(row, column, builder)
+      }
+      8 + VALIDITY
+    }
+
+    override def getNullSize: Double = 8 + VALIDITY
+  }
+
+  private class NotNullDecimalConverter(precision: Int, scale: Int) extends TypeConverter {
+    override def append(
+      row: SpecializedGetters,
+      column: Int,
+      builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
```

Comment:
Yes or it would not compile.  The Double is needed because we are estimating the size of the output data in bytes, but validity is not a full byte, so it is a bit of a hack work around.

Reply:
The return type is defined in its parent class `TypeConverter`.  I am not sure why it should be `Double`.